File: JavaScript/프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function isVPS(s) {
  const obj = { "}": "{", "]": "[", ")": "(" };
	const stack = []
	for (const alpha of s){
		if (alpha === '{' || alpha === '[' || alpha === '('){
			stack.push(alpha);
		} else {
			if (stack.length === 0) return false;
			if (stack[stack.length - 1] !== obj[alpha]) return false;
			stack.pop();
		}
	}
	return !stack.length;
}
function solution(s) {
  var answer = 0;
  len = s.length;
  s += s;
  for (let i = 0; i < len; i++) {
    if (isVPS(s.slice(i, i + len))) answer += 1;
  }
  return answer;
}

File: C++/프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.cpp
#include <string>
#include <vector>

using namespace std;

bool isVPS(string temp_s){
    vector<char> check;
    for (int i = 0; i < temp_s.size(); i++){
        char a = temp_s[i];
        if (a == ')' || a == '}' || a == ']'){
            if (check.empty()) return false;
            if (a == ')'){
                if (check[check.size() - 1] == '(') {
                    check.pop_back();
                } else return false;
            } else if (a == '}'){
                if (check[check.size() - 1] == '{') {
                    check.pop_back();
                } else return false;
            } else {
                if (check[check.size() - 1] == '[') {
                    check.pop_back();
                } else return false;
            }
        } else {
            check.push_back(a);
        }
    }
    if (!check.empty()){
        return false;
    }
    return true;
}

int solution(string s) {
    int answer = 0;
    int len = s.size();
    s += s;
    for (int i = 0; i < len; i++){
        if (isVPS(s.substr(i, len))){
            answer += 1;
        } 
    }
    return answer;
}